### PR TITLE
Close #313 Redesign of the Roles & Rules Screen/Help screen

### LINKED ISSUE #313: Redesign of the Roles & Rules Screen/Help screen
Blocks #315

### What are you proposing?

- Redesign of the roles and rules screen/help screen

### Is this ready to implement?
Yes:
#### Screenshots
Help Screen
Figma here: https://www.figma.com/file/zoB3yUb1s6WonkGmRZcRYR/MAFIA-Assignment-2?node-id=313%3A0
Screen is called 'Help screen'

Functionality and sizing remain the same as what pops up when clicking the '?' next to the players' role in the game screen, the screen has just been redesigned. 

Arrow to go back can be left the same as existing

Card heading font: Audiowide
Other Font: Roboto
Colours:
In `./Styles/colourScheme.css`
Green - #9EFF8E
Red - #F76C6C
Yellow - #E9E843
Background - #A4B8C4
Day Card Background - #C8D3D5
Night Card Background -  #293241
Black text - #2B282A
White Text - #FFFFFF

Note: All cards have a drop shadow of Blur 4, Y 4, 25% opacity 
#### Assets
Includes all the avatars, YOUR ROLE sticker, sun and moon as svg files
[MAFIA Assignment 2 (1).zip](https://github.com/Mafia-Hands/Mafia/files/6289014/MAFIA.Assignment.2.1.zip)
Also:
[Avatars without face.zip](https://github.com/Mafia-Hands/Mafia/files/6289108/Avatars.without.face.zip)
Smiley face can be found in #305 
### Redesign approval:

- [ ] Redesign approved by:

### PR DESCRIPTION
**Pull request by yzho976**
# Description
This PR fixes #[313]

Resign both in-lobby and in-game views for the rules and roles page with the new design.

# Demo
image

# Type of change:
(Delete the ones that are not relevant)
* New feature (Non-breaking change that adds functionality)

# Checklist:
* Have you merged main into your branch?
* Have you tested your changes to ensure it works as expected and does not break existing functionality?
* If applicable, please ensure sufficient tests are added that is related to the changes.
* Please ensure this PR has a `label`, is linked to an `issue` and is related to a `project`
* If new documentation is required for this change, have you created a new documentation issue that describes the documentation needed?

### CODE REVIEW

File: backend/events/game-state-events/DayStateChangeEvents.js [hunk 1]
```diff
@@ -34,10 +34,8 @@ function startDay(io, socket, mafiaGame) {
     socket.on('day-vote', () => {
         const { roomID } = socket.player;
         const room = mafiaGame.gameRoomsDict[roomID];
-        const numAlive = room.players.filter(
-            (player) => player.status === PlayerStatus.ALIVE
-        ).length;
-        const numVotes = Object.keys(room.voteHandler.daytimeVoteMap).length
+        const numAlive = room.players.filter((player) => player.status === PlayerStatus.ALIVE).length;
```
Comment (JChamberlain28) [inline backend/events/game-state-events/DayStateChangeEvents.js:37]: I assume this is an ESLint change

File: frontend/src/Components/RolesAndRules.jsx [hunk 2]
```diff
@@ -24,90 +32,126 @@ const RolesAndRules = ({ inLobby }) => {
...
-        userRole = gameState.role;
-    }
-
+    if (!inLobby) {userRole = gameState.role;}
+    
     const props = { backgroundColor: null };
+
     const classes = useStyles(props);
 
     const roles = [
```
Comment (JChamberlain28) [inline frontend/src/Components/RolesAndRules.jsx]: This is a very good PR, and I am happy to accept, but I am worried that the way the cards are done here promotes code repetition when completing issue #315. If you think that can be avoided, please suggest how.

Comment (JChamberlain28) [reply frontend/src/Components/RolesAndRules.jsx]: I am very worried this significantly increases the complexity of #315 more than needed.

**Review by JChamberlain28 — COMMENTED** [review]
Some concerns but well done

File: backend/events/game-state-events/DayStateChangeEvents.js (hunk 1 above)
Comment (yzho976) [reply backend/events/game-state-events/DayStateChangeEvents.js:37]: it is

**Review by milkyway404 — CHANGES_REQUESTED** [review]
Amazing work Evan!! I've left some information about two changes on Discord, but it's mostly around accounting for different screen sizes :)

**Review by AngelaLH — COMMENTED** [review]
Implementation matches the design really well, great work! Looks awesome on my big monitor, and should be able to work well on my small screen if scrolling is implemented

File: frontend/src/Components/RolesAndRules.jsx [hunk 3]
```diff
@@ -1,116 +1,160 @@
...
+import styles from '../Styles/RolesAndRules.module.css';
+import { GameContext, GeneralContext } from '../Context';
+import civilianImage from '../images/Civilian.svg';
+import medicImage from '../images/Medic.svg';
+import mafiaImage from '../images/Mafia.svg';
+import jesterImage from '../images/Jester.svg';
+import detectiveImage from '../images/Detective.svg';
+import sun from '../images/Sun.svg';
+import moon from '../images/Moon.svg';
+import yourRole from '../images/Your Role.svg'
```
Comment (khal745) [inline frontend/src/Components/RolesAndRules.jsx:13]: Nit: Probably better to have file names without spaces in them, but shouldn't be an issue here

File: frontend/src/Styles/RolesAndRules.module.css [hunk 4]
```diff
@@ -1,38 +1,100 @@
...
-    padding: 0.5em;
-    border-radius: 10px 10px 0 0;
-    font-weight: 500;
-}
+@import './colourScheme.css';
+@import url('https://fonts.googleapis.com/css?family=Audiowide');
+
+.pageContainer {
+    align-items: center;
+    background-color: #A4B8C4!important;
```
Comment (khal745) [inline frontend/src/Styles/RolesAndRules.module.css]: Nit: This is already defined in the `colorScheme.css` file as cadetBlueCrayola, and you used this in `ModulMUI.module.css`, but easy enough to fix up later.

**Review by khal745 — DISMISSED** [review]
Fantastic work with this, awesome job! Matches the design well! Some minor issues with small screens, but I think its readable and worth adding in. Some small nitpick comments that might be worth changing in the upcoming PRs, but nothing that should stop this getting merged in, in my opinion.

File: frontend/src/Components/RolesAndRules.jsx (hunk 2 above)
Comment (JChamberlain28) [reply frontend/src/Components/RolesAndRules.jsx]: It is impractical to change how the code is here given time constraints. #315 is still possible to do, just more complex, so this comment is resolved

**Review by JChamberlain28 — DISMISSED** [review]
I have tested this on my computer. I have minor display issues due to my screen resolution and high DPI settings. As this is not evident on a lot of computers, I believe this is acceptable and a bug could be opened if we found this to be an issue in future. Good work, this follows the design accurately and I approve.

**Review by milkyway404 — APPROVED** [review]
Thank you for the amazing work! I've tested on my local and it's still a bit squished but I'm happy to approve for now and implement the changes later if necessary.

**Review by AngelaLH — APPROVED** [review]
Happy to approve for now as it matches the design closely, great work!

**Comment by khal745** [comment]
While the contributor guidelines state that the author should merge their own PR, due to some time zone differences and dependencies on follow up work we merged this PR in on his behalf

**Pull request merged**